crates/components/aead/src/aes_gcm/mod.rs

```diff
@@ -423,294 +423,306 @@ impl<Ctx: Context> Aead for MpcAesGcm<Ctx> {
-    //    async fn test_aes_gcm_verify_tag() {
-    //        let key = vec![0u8; 16];
-    //        let iv = vec![0u8; 4];
-    //        let explicit_nonce = vec![0u8; 8];
-    //        let plaintext = vec![1u8; 32];
-    //        let aad = vec![2u8; 12];
-    //        let ciphertext = reference_impl(&key, &iv, &explicit_nonce, &plaintext, &aad);
-    //
-    //        let len = ciphertext.len();
-    //
-    //        let (mut leader, mut follower) = setup_pair(key.clone(), iv.clone()).await;
-    //
-    //        tokio::try_join!(
-    //            leader.verify_tag(explicit_nonce.clone(), ciphertext.clone(), aad.clone()),
-    //            follower.verify_tag(explicit_nonce.clone(), ciphertext.clone(), aad.clone())
-    //        )
-    //        .unwrap();
-    //
-    //        //Corrupt tag.
-    //        let mut corrupted = ciphertext.clone();
-    //        corrupted[len - 1] -= 1;
-    //
-    //        let (leader_res, follower_res) = tokio::join!(
-    //            leader.verify_tag(explicit_nonce.clone(), corrupted.clone(), aad.clone()),
-    //            follower.verify_tag(explicit_nonce.clone(), corrupted, aad.clone())
-    //        );
-    //
-    //        assert_eq!(leader_res.unwrap_err().kind(), ErrorKind::Tag);
-    //        assert_eq!(follower_res.unwrap_err().kind(), ErrorKind::Tag);
-    //    }
+    use super::*;
+    use crate::{
+        aes_gcm::{mock::create_mock_aes_gcm_pair, AesGcmConfigBuilder, Role},
+        Aead,
+    };
+    use ::aes_gcm::{aead::AeadInPlace, Aes128Gcm, NewAead, Nonce};
+    use error::ErrorKind;
+    use mpz_common::executor::STExecutor;
+    use mpz_garble::{protocol::deap::mock::create_mock_deap_vm, Memory};
+    use serio::channel::MemoryDuplex;
+    use std::sync::Once;
+
+    // We need to configure the rayon global thread pool (which is used downstream in mpz-common)
+    // to increase the maximum number of rayon threads which are run in parallel in order to prevent a
+    // deadlock in these unit tests when running in our github ci.
+    //
+    // c.f. https://github.com/tlsnotary/tlsn/issues/548
+    static INIT_RAYON_THREADPOOL: Once = Once::new();
+
+    fn reference_impl(
+        key: &[u8],
+        iv: &[u8],
+        explicit_nonce: &[u8],
+        plaintext: &[u8],
+        aad: &[u8],
+    ) -> Vec<u8> {
+        let cipher = Aes128Gcm::new_from_slice(key).unwrap();
+        let nonce = [iv, explicit_nonce].concat();
+        let nonce = Nonce::from_slice(nonce.as_slice());
+
+        let mut ciphertext = plaintext.to_vec();
+        cipher
+            .encrypt_in_place(nonce, aad, &mut ciphertext)
+            .unwrap();
+
+        ciphertext
+    }
+
+    async fn setup_pair(
+        key: Vec<u8>,
+        iv: Vec<u8>,
+    ) -> (
+        MpcAesGcm<STExecutor<MemoryDuplex>>,
+        MpcAesGcm<STExecutor<MemoryDuplex>>,
+    ) {
+        INIT_RAYON_THREADPOOL.call_once(|| {
+            rayon::ThreadPoolBuilder::new()
+                .num_threads(9)
```

Comment:
This is the actual fix. Maybe it is worth converting 9 into a CONSTANT and documenting in this file why you picked 9 as magic number.

Reply:
I will wait a little bit, because maybe the fix is to ditch rayon, instead. We will discuss today.